server/rw_wrapper.go

```diff
@@ -0,0 +1,63 @@
+package server
+
+import (
+	"bytes"
+	"compress/gzip"
+	"net/http"
+
+	"github.com/avenga/couper/handler"
+)
+
+var _ http.ResponseWriter = &RWWrapper{}
+
+// RWWrapper wraps the <http.ResponseWriter>.
+type RWWrapper struct {
+	rw    http.ResponseWriter
+	useGZ bool
+}
+
+// NewRWWrapper creates a new RWWrapper object.
+func NewRWWrapper(rw http.ResponseWriter, useGZ bool) *RWWrapper {
+	return &RWWrapper{
+		rw:    rw,
+		useGZ: useGZ,
+	}
+}
+
+// Header wraps the Header method of the <http.ResponseWriter>.
+func (w *RWWrapper) Header() http.Header {
+	return w.rw.Header()
+}
+
+// Write wraps the Write method of the <http.ResponseWriter>.
+func (w *RWWrapper) Write(p []byte) (int, error) {
+	if w.useGZ {
```

Comment:
For readability, inverse option check and add a return write would remove the gzip related indent.